chain/actors/builtin/datacap/actor.go.template

```diff
@@ -0,0 +1,54 @@
+package datacap
+
+import (
+    "golang.org/x/xerrors"
+
+    "github.com/filecoin-project/go-address"
+    "github.com/filecoin-project/go-state-types/abi"
+    actorstypes "github.com/filecoin-project/go-state-types/actors"
+    builtin{{.latestVersion}} "github.com/filecoin-project/go-state-types/builtin"
+    "github.com/filecoin-project/go-state-types/cbor"
+
+    "github.com/filecoin-project/lotus/chain/actors"
+    "github.com/filecoin-project/lotus/chain/actors/adt"
+    "github.com/filecoin-project/lotus/chain/types"
+)
+
+var (
+	Address = builtin9.DatacapActorAddr
+	Methods = builtin9.MethodsDatacap
+)
+
+func Load(store adt.Store, act *types.Actor) (State, error) {
+	if name, av, ok := actors.GetActorMetaByCode(act.Code); ok {
+       if name != actors.DatacapKey {
+          return nil, xerrors.Errorf("actor code is not datacap: %s", name)
+       }
+
+       switch av {
+            {{range .versions}}
+                case actorstypes.Version{{.}}:
+                     return load{{.}}(store, act.Head)
+            {{end}}
+       }
+	}
+
+	return nil, xerrors.Errorf("unknown actor code %s", act.Code)
+}
+
+func MakeState(store adt.Store, av actorstypes.Version, governor address.Address, bitwidth uint64) (State, error) {
+	switch av {
+{{range .versions}}
+	case actorstypes.Version{{.}}:
+		return make{{.}}(store, governor, bitwidth)
+{{end}}
```

Comment:
Let's have a `default` here that fails saying datacap is only supported for v9 and up.